integration/testrunner/db.go

```diff
@@ -16,45 +16,65 @@ package testrunner
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"testing"
+	"time"
 
 	"cloud.google.com/go/bigquery"
+	logpb "cloud.google.com/go/logging/apiv2/loggingpb"
 	"github.com/sethvargo/go-retry"
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/googleapis/cloud/audit"
+	"google.golang.org/protobuf/encoding/protojson"
 )
 
 // queryIfAuditLogExists queries the DB and checks if audit log contained in the query exists or not.
-func queryIfAuditLogExists(ctx context.Context, tb testing.TB, query *bigquery.Query, expectedNum int64) (bool, error) {
+func queryAuditLog(ctx context.Context, tb testing.TB, query *bigquery.Query) ([]*logpb.LogEntry, error) {
 	tb.Helper()
-
+	time.Sleep(20 * time.Second)
 	job, err := query.Run(ctx)
 	if err != nil {
-		return false, fmt.Errorf("failed to run query: %w", err)
+		tb.Logf("failed to run query: %s", err.Error())
+		return nil, fmt.Errorf("failed to run query: %w", err)
 	}
 
 	if status, err := job.Wait(ctx); err != nil {
-		return false, fmt.Errorf("failed to wait for query: %w", err)
+		tb.Logf("failed to wait for query: %s", err.Error())
+		return nil, fmt.Errorf("failed to wait for query: %w", err)
 	} else if err = status.Err(); err != nil {
-		return false, fmt.Errorf("query failed: %w", err)
+		return nil, fmt.Errorf("query failed: %w", err)
 	}
-
 	it, err := job.Read(ctx)
 	if err != nil {
-		return false, fmt.Errorf("failed to read job: %w", err)
-	}
-
-	var row []bigquery.Value
-	if err = it.Next(&row); err != nil {
-		return false, fmt.Errorf("failed to get next row: %w", err)
+		tb.Logf("failed to read job: %s", err.Error())
+		return nil, fmt.Errorf("failed to read job: %w", err)
 	}
-
-	// Check if the matching row count is equal to expected, if yes, then the audit log exists.
-	tb.Logf("Found %d matching rows", row[0])
-	result, ok := row[0].(int64)
-	if !ok {
-		return false, fmt.Errorf("error converting query results to integer value (got %T)", row[0])
+	var results []*logpb.LogEntry
+	for {
+		var row []bigquery.Value
+		err := it.Next(&row)
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		if err != nil {
+			tb.Logf("failed to get next row")
+			return nil, fmt.Errorf("failed to get next row: %w", err)
+		}
+		tb.Logf("%s", row)
+		pbJSON := &logpb.LogEntry{}
+		value, ok := row[0].(string)
+		if !ok {
+			tb.Logf("error converting query results to string")
+			return nil, fmt.Errorf("error converting query results to string (got %T)", value[0])
+		}
+		if err := protojson.Unmarshal([]byte(value), pbJSON); err != nil {
+			tb.Logf("ignoring error: %s as this behavior is expected", err.Error())
```

Comment:
Because it's expected, is the json string has fields with null, there will be an error, but the json can still be Unmarshal into `pbJson`

Reply:
protojson does not support it, right? As it indicates here: https://github.com/golang/protobuf/issues/1313. In addition, can you add a comment to clarify why we ignore the error here.

Reply:
Yeah, it's protojson's issue, but it doesn't affect our stuffs.

Reply:
Can you add a link of https://github.com/golang/protobuf/issues/1313 in the comment to clarify it and it would be better to add comment just above the `tb.Logf("ignoring error: %s as this behavior is expected", err.Error())`, now it looks a little bit confusing.

Reply:
Done

Reply:
To clarify, as https://github.com/golang/protobuf/issues/1313 mentioned, the protojson package only supports null values as direct field values, not if they are elements in JSON arrays or values in JSON maps. Can you help update the comments ?

Reply:
done